File: frontend/src/utils/depositRate.ts
import type { DepositRate } from "@/types/payment";

export function toDepositRate(rate: number): DepositRate {
  switch (rate) {
    case 10:
      return "TEN";
    case 20:
      return "TWENTY";
    case 30:
      return "THIRTY";
    case 40:
      return "FORTY";
    case 50:
      return "FIFTY";
    default:
      return "TEN";
  }
}

File: backend/src/main/java/com/eatsfine/eatsfine/domain/store/enums/DepositRate.java
package com.eatsfine.eatsfine.domain.store.enums;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Getter;
import lombok.RequiredArgsConstructor;

@Schema(
        description = "예약금 비율",
        allowableValues = {"10%", "20%", "30%", "40%", "50%"}
)
@Getter
@RequiredArgsConstructor
public enum DepositRate {
    TEN(10),
    TWENTY(20),
    THIRTY(30),
    FORTY(40),
    FIFTY(50);

    private final int percent;
}
